react-app/src/components/ListItem/index.js

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import ActionIcon from 'components/ActionIcon';
+import Button from 'components/Button';
+
+import styles from './index.module.scss';
+
+export default class ListItem extends React.PureComponent {
+  static propTypes = {
+    /**
+     * The date of the entry
+     */
+    date: PropTypes.string.isRequired,
+
+    /**
+     * The text of the entry
+     */
+    text: PropTypes.string.isRequired,
+
+    /**
+     * The function to call when edit button is tapped
+     */
+    onEdit: PropTypes.func.isRequired,
+
+    /**
+     * The function to call when delete button is tapped
+     */
+    onDelete: PropTypes.func.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = { expanded: false }
+  }
+
+  _handleEdit = () => this.props.onEdit(this.props.date);
+  _handleDelete = () => this.props.onDelete(this.props.date);
+
+  _toggleExpanded = () => this.setState({ expanded: !this.state.expanded });
+
+  _getActionIcon() {
+    const icon = this.state.expanded ? ActionIcon.COLLAPSE : ActionIcon.EXPAND;
+
+    return <ActionIcon
+      icon={icon}
+      onClick={this._toggleExpanded}
+    />;
+  }
+
+  _getEditButton() {
+    return <Button onClick={this._handleEdit}>Edit</Button>;
+  }
+
+  _getDeleteButton() {
+    return <Button onClick={this._handleDelete}>Delete</Button>;
+  }
+
+  render() {
+    const icon = this._getActionIcon();
+    const editButton = this.state.expanded ? this._getEditButton() : null;
```

Comment:
I think that instead of using a ternary operator I'd write:

```js
  _getEditButton() {
    if (!this.state.expanded) return null;
    return <Button onClick={this._handleEdit}>Edit</Button>;
  }

  render() {
    const editButton = this._getEditButton();
```

But that's a matter of personal preference I guess.
Nothing to change here, I'm just sharing an opinion.